File: vol1/ch11/LeetCode523.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        int n=nums.size();
        if (n<2) return false;
        unordered_map<int,int> hmap;
        hmap[0]=-1;
        int presum=0;
        for (int j=0;j<n;j++) {
            presum=(presum+nums[j]) % k;
            if (hmap.count(presum)>0) {
                if (j-hmap[presum]>=2) return true;
            }
            else hmap[presum]=j;
        }
        return false;
    }
};

File: vol1/ch11/LeetCode523.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        n=len(nums)
        if n<2:return False
        hmap={}
        hmap[0]=-1
        presum=0 
        for j in range(0,n):
            presum=(presum+nums[j])%k
            if presum in hmap:
                if j-hmap[presum]>=2:return True
            else:hmap[presum]=j
        return False
